tests/api_tests/test_add_documents_common.py

```diff
@@ -0,0 +1,115 @@
+            {
+                "indexName": cls.structured_text_index_name,
+                "type": "structured",
+                "model": "sentence-transformers/all-MiniLM-L6-v2",
+                "allFields": [
+                    {"name": "title", "type": "text", "features": ["filter", "lexical_search"]},
+                    {"name": "content", "type": "text", "features": ["filter", "lexical_search"]},
+                ],
+                "tensorFields": ["title", "content"],
+            },
+            {
+                "indexName": cls.structured_filter_index_name,
+                "type": "structured",
+                "model": "sentence-transformers/all-MiniLM-L6-v2",
+                "allFields": [
+                    {"name": "field_a", "type": "text", "features": ["filter", "lexical_search"]},
+                    {"name": "field_b", "type": "text", "features": ["filter"]},
+                    {"name": "str_for_filtering", "type": "text", "features": ["filter"]},
+                    {"name": "int_for_filtering", "type": "int", "features": ["filter"]},
+                    {"name": "long_field_1", "type": "long", "features": ["filter"]},
+                    {"name": "double_field_1", "type": "double", "features": ["filter"]},
+                    {"name": "array_long_field_1", "type": "array<long>", "features": ["filter"]},
+                    {"name": "array_double_field_1", "type": "array<double>", "features": ["filter"]}
+                ],
+                "tensorFields": ["field_a", "field_b"],
+            },
+            {
+                "indexName": cls.structured_image_index_name,
+                "type": "structured",
+                "model": "open_clip/ViT-B-32/openai",
+                "allFields": [
+                    {"name": "title", "type": "text", "features": ["filter", "lexical_search"]},
+                    {"name": "content", "type": "text", "features": ["filter", "lexical_search"]},
+                    {"name": "image_content", "type": "image_pointer"},
+                    {"name": "image_field_1", "type": "image_pointer"},
+                ],
+                "tensorFields": ["title", "image_content", "image_field_1"],
+            }
+        ])
+
+        cls.create_indexes([
+            {
+                "indexName": cls.unstructured_text_index_name,
+                "type": "unstructured",
+                "model": "sentence-transformers/all-MiniLM-L6-v2",
+            },
+            {
+                "indexName": cls.unstructured_image_index_name,
+                "type": "unstructured",
+                "model": "open_clip/ViT-B-32/openai"
+            }
+        ])
+
+        cls.indexes_to_delete = [cls.structured_image_index_name, cls.structured_filter_index_name,
+                                 cls.structured_text_index_name, cls.unstructured_image_index_name,
+                                 cls.unstructured_text_index_name]
+
+    def test_add_documents_for_private_images(self):
```

Comment:
Could we test these on existing indexes as there's time overhead for creating indexes?

Reply:
If I test on existing indexes, these tests will be in two separate structured and unstructured indexes t. I want to combine these tests so I create this new class